### PR TITLE
refactor(parser): remove unused name_in_lower_case

### PR DESCRIPTION
**Pull request by lmatz**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

We want to be case-sensitive.
follow up of #9086 

## Checklist For Contributors

~- [ ] I have written necessary rustdoc comments~
~- [ ] I have added necessary unit tests and integration tests~
~- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).~
~- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)~
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: src/connector/src/parser/maxwell/simd_json_parser.rs [hunk 1]
```diff
@@ -128,8 +122,8 @@ mod tests {
     #[tokio::test]
     async fn test_json_parser() {
         let descs = vec![
-            SourceColumnDesc::simple("ID", DataType::Int32, 0.into()),
-            SourceColumnDesc::simple("NAME", DataType::Varchar, 1.into()),
+            SourceColumnDesc::simple("id", DataType::Int32, 0.into()),
+            SourceColumnDesc::simple("name", DataType::Varchar, 1.into()),
```
Comment (tabVersion) [inline src/connector/src/parser/maxwell/simd_json_parser.rs:126]: the test is intentional. for the mismatch between column name and json keys

**Review by tabVersion — APPROVED** [review]
rest LGTM

File: src/connector/src/parser/maxwell/simd_json_parser.rs (hunk 1 above)
Comment (lmatz) [reply src/connector/src/parser/maxwell/simd_json_parser.rs:126]: Do you mean that we need both `match` and `mismatch` test cases?

Without the changes, https://github.com/risingwavelabs/risingwave/pull/9106/files#diff-9ba8fcf927d4f456ea34620beb6dfe24ffe28683b2f6b40bd8e2aa76a1955377R153 currently cannot pass

Comment (tabVersion) [reply src/connector/src/parser/maxwell/simd_json_parser.rs:126]: https://github.com/risingwavelabs/risingwave/pull/9086 should handle such cases, column name `Name`, `NAme` should be able to match `name` in json

**Review by fuyufjh — APPROVED** [review]

**Pull request merged**